test/validate/validate.go

```diff
@@ -154,77 +167,121 @@ func (v *Validator) validateIPs(ctx context.Context, stateFileIps stateFileIpsFu
 		if err != nil {
 			return errors.Wrapf(err, "failed to exec into privileged pod - %s", podName)
 		}
-		filePodIps, err := stateFileIps(result)
+		cnsPodIPs, err := stateFileIps(result)
 		if err != nil {
 			return errors.Wrapf(err, "failed to get pod ips from state file")
 		}
-		if len(filePodIps) == 0 && v.restartCase {
+		if len(cnsPodIPs) == 0 && v.restartCase {
 			log.Printf("No pods found on node %s", nodes.Items[index].Name)
 			continue
 		}
 		// get the pod ips
-		podIps := getPodIPsWithoutNodeIP(ctx, v.clientset, nodes.Items[index])
-
-		check := compareIPs(filePodIps, podIps)
+		observedPodIPs := getPodIPsWithoutNodeIP(ctx, v.clientset, nodes.Items[index])
 
-		if !check {
+		if err := compareIPs(cnsPodIPs, observedPodIPs); err != nil {
 			return errors.Wrapf(errors.New("State file validation failed"), "for %s on node %s", checkType, nodes.Items[index].Name)
 		}
 	}
 	log.Printf("State file validation for %s passed", checkType)
 	return nil
 }
 
-func (v *Validator) validateDualStackNodeProperties(ctx context.Context) error {
-	log.Print("Validating Dualstack Overlay Node properties")
-	nodes, err := acnk8s.GetNodeListByLabelSelector(ctx, v.clientset, nodeSelectorMap[v.os])
-	if err != nil {
-		return errors.Wrapf(err, "failed to get node list")
-	}
+func validateNodeProperties(nodes *corev1.NodeList, labels map[string]string, expectedIPCount int) error {
+	log.Print("Validating Node properties")
 
 	for index := range nodes.Items {
 		nodeName := nodes.Items[index].ObjectMeta.Name
 		// check nodes status;
 		// nodes status should be ready after cluster is created
 		nodeConditions := nodes.Items[index].Status.Conditions
 		if nodeConditions[len(nodeConditions)-1].Type != corev1.NodeReady {
-			return errors.Wrapf(err, "node %s status is not ready", nodeName)
+			return errors.Errorf("node %s status is not ready", nodeName)
 		}
 
 		// get node labels
 		nodeLabels := nodes.Items[index].ObjectMeta.GetLabels()
 		for key := range nodeLabels {
-			if label, ok := dualstackOverlayNodeLabels[key]; ok {
+			if label, ok := labels[key]; ok {
 				log.Printf("label %s is correctly shown on the node %+v", key, nodeName)
 				if label != overlayClusterLabelName {
-					return errors.Wrapf(err, "node %s overlay label name is wrong; expected label:%s but actual label:%s", nodeName, overlayClusterLabelName, label)
+					return errors.Errorf("node %s overlay label name is wrong; expected label:%s but actual label:%s", nodeName, overlayClusterLabelName, label)
 				}
 			}
 		}
 
-		// check if node has two internal IPs(one is IPv4 and another is IPv6)
+		// check if node has corret number of internal IPs
```

Comment:
`corret` -> `correct`

Reply:
Sure, I'll "corret" that 😃
<crickets..>